pkg/keycloak/realm/phaseHandler.go

```diff
@@ -205,6 +205,8 @@ func (ph *phaseHandler) reconcileUser(kcUser, specUser *v1alpha1.KeycloakUser, r
 		if u == nil {
 			return errors.New("failed to find user " + specUser.Email)
 		}
+		specUser.ID = u.ID
```

Comment:
@philbrookes btw, I think this fixes https://issues.jboss.org/browse/INTLY-2799

Reply:
aha nice!